flink-libraries/flink-cep/src/main/java/org/apache/flink/cep/nfa/NFA.java

```diff
@@ -322,6 +336,64 @@ public void resetNFAChanged() {
 
 		}
 
+		Set<T> discardEvents = new HashSet<>();
```

Comment:
move the whole added block to a new method. The method `process` is already long.